Move file reading docs to cairo (#1997)
foundry-rs/starknet-foundry: docs/src/appendix/snforge-library/fs.md

Module containing functions for interacting with the filesystem.

## `File`
```
trait FileTrait {
    fn new(path: ByteArray) -> File;
}
```

## `FileParser<T>`
```rust

> trait FileParser<T, impl TSerde: Serde<T>> {

Reviewer: ```suggestion
trait FileParser<T, +Serde<T>> {
```
https://github.com/foundry-rs/starknet-foundry/issues/2010

    fn parse_txt(file: @File) -> Option<T>;
    fn parse_json(file: @File) -> Option<T>;
}
```

## `read_txt` & `read_json`
```rust
fn read_txt(file: @File) -> Array<felt252>;
fn read_json(file: @File) -> Array<felt252>;
```

> ## File format

Reviewer 2: As an end goal we probably would like to put those in cairo, yup?

Author: I thought of a guide here actually, but both would work

Some rules have to be checked when providing a file for snforge, in order for correct parsing behavior.

> Different ones apply for JSON and plain text files.

Reviewer: ```suggestion
The rules differ between JSON and plain text files.
```

Author: ah tomato tomato

### Plain text files
- Elements have to be separated with newlines
- Elements have to be either:
  - integers in range of `[0, P)` where P is [`Cairo Prime`](https://book.cairo-lang.org/ch02-02-data-types.html?highlight=prime#felt-type) either in decimal or `0x` prefixed hex format
  - single line short strings (`felt252`) of length `<=31` surrounded by `''` i.e., `'short string'`, new lines can be used with `\n` and `'` with `\'`
  - single line strings (`ByteArray`) surrounded by `""` i.e., `"very very very very loooooong string"`, new lines can be used with `\n` and `"` with `\"`

### JSON files
- Elements have to be either:
  - integers in range of `[0, P)` where P is [`Cairo Prime`](https://book.cairo-lang.org/ch02-02-data-types.html?highlight=prime#felt-type)
  - single line strings (`ByteArray`) i.e. `"very very very very loooooong string"`, new lines can be used with `\n` and `"` with `\"`
  - array of integers or strings fulfilling the above conditions

> ⚠️ **Warning**
>

> > A JSON object is an unordered data structure. To give it an order, the values in the array are sorted alphabetically by JSON keys. Therefore, the values in the array are sorted alphabetically by JSON key.

Reviewer: ```suggestion
> A JSON object is an unordered data structure. To make reading JSONs deterministic, the values are read from the JSON in an order that is alphabetical in respect to JSON keys.
```
